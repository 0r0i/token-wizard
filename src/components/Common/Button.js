import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
