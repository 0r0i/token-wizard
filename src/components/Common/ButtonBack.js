import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
