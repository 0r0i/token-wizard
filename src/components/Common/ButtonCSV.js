import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
