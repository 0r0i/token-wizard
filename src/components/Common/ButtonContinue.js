import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
