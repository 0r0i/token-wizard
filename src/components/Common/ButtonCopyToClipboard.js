import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
