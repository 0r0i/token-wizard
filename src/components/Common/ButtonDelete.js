import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
