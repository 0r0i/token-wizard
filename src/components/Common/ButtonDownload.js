import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
