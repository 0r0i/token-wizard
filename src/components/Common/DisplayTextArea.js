import React from 'react'
import { ButtonCopyToClipboard } from '../Common/ButtonCopyToClipboard'

export const DisplayTextArea = ({ title, value, key = '', description = '' }) => {
  return (
    <div key={key ? key : ''} className="pb-DisplayTextArea">
      <h3 className="pb-DisplayTextArea_Title">
        {title} <ButtonCopyToClipboard value={value} title={title} />
      </h3>
      <pre className="pb-DisplayTextArea_Content">{value}</pre>
      {description ? <p className="pb-DisplayTextArea_Description">{description}</p> : null}
    </div>
  )
}
