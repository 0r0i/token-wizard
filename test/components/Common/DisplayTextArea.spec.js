import React from 'react'
import { DisplayTextArea } from '../../../src/components/Common/DisplayTextArea'
import { Form } from 'react-final-form'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'
import { FORM_ERROR } from 'final-form'

configure({ adapter: new Adapter() })

describe('Error ', () => {
  const label = 'Label'
  const description = 'Description'
  const key = 'Key'
  const value='Value'
  it(`should render Error component`, () => {

    const wrapper = mount(
      <DisplayTextArea
        key={key}
        label={label}
        description={description}
        value={value}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
