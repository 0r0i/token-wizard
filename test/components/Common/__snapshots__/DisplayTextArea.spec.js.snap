// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error  should render Error component 1`] = `
<DisplayTextArea
  description="Description"
  label="Label"
  value="Value"
>
  <div
    className="item"
  >
    <div>
      <div
        className="display-container"
      >
        <p
          className="label"
        >
          Label
        </p>
      </div>
      <div
        className="copy-area-container"
      >
        <btn
          className="copy"
          data-clipboard-action="copy"
          data-clipboard-text="Value"
        />
      </div>
    </div>
    <pre>
      Value
    </pre>
    <p
      className="description"
    >
      Description
    </p>
  </div>
</DisplayTextArea>
`;
